Sources/SwifterAuth.swift

```diff
@@ -114,6 +115,63 @@ public extension Swifter {
             }
         }, failure: failure)
     }
+  
+    func authorizeSSO(success: SSOTokenSuccessHandler?,
+                      failure: FailureHandler? = nil) {
+        guard let client = client as? SwifterAppProtocol else {
+            let error = SwifterError(message: "SSO not supported AppOnly client",
+                                     kind: .invalidClient)
+            failure?(error)
+            return
+        }
+        
+        let urlScheme = "swifter-\(client.consumerKey)"
+        
+        let nc = NotificationCenter.default
+        self.swifterCallbackToken = nc.addObserver(forName: .swifterSSOCallback, object: nil, queue: .main) { notification in
+            self.swifterCallbackToken = nil
+            let url = notification.userInfo![CallbackNotification.optionsURLKey] as! URL
+            guard url.scheme == urlScheme else { return }
+            
+            let isCanceled = url.host == nil
+            if isCanceled {
+                let error = SwifterError(message: "User cancelled login from Twitter App",
+                                         kind: .cancelled)
+                failure?(error)
+            } else {
+                let params = url.queryParamsForSSO
+                let secret = params["secret"]!
+                let token = params["token"]!
+                let credentialToken = Credential.OAuthAccessToken(key: token, secret: secret)
+                self.client.credential = Credential(accessToken: credentialToken)
+                success?(credentialToken)
+            }
+        }
+        
+        let url = URL(string: "twitterauth://authorize?consumer_key=\(client.consumerKey)&consumer_secret=\(client.consumerSecret)&oauth_callback=\(urlScheme)")!
+        UIApplication.shared.open(url, options: [:], completionHandler: { (success) in
+            if !success {
+                let error = SwifterError(message: "Cannot open twitter app",
+                                         kind: .noTwitterApp)
+                failure?(error)
+            }
+        })
+    }
+    
+    class func application(_ app: UIApplication, open url: URL, options: [UIApplication.OpenURLOptionsKey : Any]) -> Bool {
```

Comment:
Don't pass in all the arguments with UIKit types (i.e. UIApplication). Again, we need to be platform-agnostic with the callback. AppKit will not recognize any of this. Instead, we should have something like

```swift
class func handleOpenUrl(_ url: URL, callbackUrl: URL, source: String? = nil) -> Bool { ... }
```

Reply:
Fixed!
ba79795